cot-cli/src/utils.rs

```diff
@@ -67,20 +70,373 @@ impl StatusType {
+            .map(|p| Self::from_cargo_toml_path(&p))
+            .transpose()
+    }
+
+    pub(crate) fn find_cargo_toml(starting_dir: &Path) -> Option<PathBuf> {
+        let mut current_dir = starting_dir;
+
+        loop {
+            let candidate = current_dir.join("Cargo.toml");
+            if candidate.exists() {
+                return Some(candidate);
+            }
 
-        match current_dir.parent() {
-            Some(parent) => current_dir = parent,
-            None => break,
+            match current_dir.parent() {
+                Some(parent) => current_dir = parent,
+                None => break,
+            }
         }
+
+        None
     }
 
-    None
+    #[allow(unused)]
+    pub(crate) fn get_packages(&self) -> Vec<String> {
+        self.package_manifests.keys().cloned().collect()
+    }
+
+    pub(crate) fn get_package_manifest(&self, package_name: &str) -> Option<&Manifest> {
+        self.package_manifests
+            .get(package_name)
+            .map(|m| &m.manifest)
+    }
+
+    pub(crate) fn get_package_manifest_by_path(&self, package_path: &Path) -> Option<&Manifest> {
+        let mut package_path = package_path
+            .canonicalize()
+            .context("unable to canonicalize path")
+            .ok()?;
+
+        if package_path.is_dir() {
+            package_path = package_path.join("Cargo.toml");
+        }
+
+        self.package_manifests.values().find_map(|m| {
+            if m.path == package_path {
+                Some(&m.manifest)
+            } else {
+                None
+            }
+        })
+    }
+
+    pub(crate) fn get_manifest_path(&self, package_name: &str) -> Option<&Path> {
+        self.package_manifests
+            .get(package_name)
+            .map(|m| m.path.as_path())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+
+    use std::io::Write;
+
+    use cargo;
+    use cargo::ops::NewProjectKind;
+
+    const WORKSPACE_STUB: &'static str = "[workspace]\nresolver = \"3\"";
+
+    fn make_workspace_package(path: PathBuf, packages: u8) -> anyhow::Result<()> {
+        let workspace_cargo_toml = path.join("Cargo.toml");
+        std::fs::write(workspace_cargo_toml, WORKSPACE_STUB)?;
+
+        for i in 0..packages {
+            let package_path = path.join(format!("cargo-test-crate-{i}"));
+            make_package(package_path)?;
+        }
+
+        Ok(())
+    }
+    fn make_package(path: PathBuf) -> anyhow::Result<()> {
+        let new_options = cargo::ops::NewOptions {
+            version_control: None,
+            kind: NewProjectKind::Lib,
+            auto_detect_kind: false,
+            path: path.clone(),
+            name: None,
+            edition: None,
+            registry: None,
+        };
+        let global_context = cargo::GlobalContext::default()?;
+        match path.exists() {
+            true => cargo::ops::init(&new_options, &global_context).map(|_| ()),
+            false => cargo::ops::new(&new_options, &global_context),
```

Comment:
As discussed on another channel, I think we should avoid adding 60% more dependencies (as measured by `cargo build --all-features --tests`) if there's no strong advantage. We already run Cargo via command line in [another test](https://github.com/cot-rs/cot/blob/master/cot-cli/tests/new_project.rs#L40-L55), and I think it's fine to do that here, too.